### DISCUSSION TITLE
[New Feature] Explicitly definable sequences for Tiberium Spawners and Veinhole Monster

### DISCUSSION THREAD

**Discussion by MNOrac** [Vinifera-Developers/Vinifera#1381] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

`veinhole.tem` is 12 frames long (plus shadows), and while it used these frames fairly effectively, the limitation makes it difficult to modify or modernize the asset.

Frame 1 is used while the VHM is dormant
Frames 2-7 are used as it twitches (idle? spawning veins? unclear)
Frame 8-11 are used when it spits out gas
Frame 12 is used as it dies

Tiberium Trees and BIGBLUE spawners have similarly inflexible sequences, though without gas-spitting or death frames involved. 

### Proposed Code (optional)

These sequences could be made explicit, possibly applying the same naming conventions as units (though this is potentially a weird pattern).

`DormantFrames`, new keys for while the overlay is inactive.
`IdleFrames` existing keys for while it is twitching (if this is what it's doing, I cannot tell by looking).
`SpawningFrames` new keys for when the tree/vhm is spawning new tib/veins.
`FiringFrames` existing keys for when the vhm is spitting out gas.
`DeathFrames` existing keys for as the vhm dies.

### Additional Context

![VHM](https://user-images.githubusercontent.com/10970765/166104107-2bad7b2e-f2ef-4b62-8cb5-3e781e48b465.gif)

**Comment by Crimsonum** [comment]
Veinhole animations are unrelated to vein growth. Still, `SpawningFrames` could be interesting, perhaps the simplest way to implement it would be playing the spawn frames every `VeinholeGrowthRate` frames.

Also, slight correction to the anim sequence: The veinhole alternates between 1st and 4th frames as idle states (mouth closed/open, respectively), with 2nd and 3rd frames as transition frames. Sometimes it plays from 1st frame to 7th, then cuts back to 1st, making it look like it's breathing. The 12th frame is indeed used only when the veinhole dies, at the end of the attack/gas-spawning-sequence, after the veinhole quite literally draws its last breath. Not that any of this matters for the feature request, but perhaps it helps when identifying/de-hardcoding the veinhole code.
